packages/clay-core/stories/FocusTrap.stories.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * SPDX-FileCopyrightText: © 2022 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayButton from '@clayui/button';
+import ClayCard from '@clayui/card';
+import React, {useRef, useState} from 'react';
+
+import {FocusTrap} from '../src/focus-trap';
+
+export default {
+	component: FocusTrap,
+	title: 'Design System/Components/FocusTrap',
+};
+
+export const Default = () => {
+	const [active, setActive] = useState(false);
+	const activateTrapButton = useRef<HTMLButtonElement>(null);
+
+	const onActivateFocusTrap = () => {
+		setActive(true);
+	};
+
+	const onDeactivateFocusTrap = () => {
+		setActive(false);
+
+		if (activateTrapButton.current) {
+			activateTrapButton.current.focus();
+		}
+	};
+
+	return (
+		<>
+			<ClayButton onClick={onActivateFocusTrap} ref={activateTrapButton}>
+				Activate trap
+			</ClayButton>
+			{active && (
+				<FocusTrap active={active} onDeactivate={onDeactivateFocusTrap}>
+					<ClayCard className="mt-4 p-4">
+						<ClayButton displayType="link">Button 1</ClayButton>
+						<ClayButton displayType="link">Button 2</ClayButton>
+						<div className="mt-4">
+							<ClayButton onClick={onDeactivateFocusTrap}>
+								Leave trap
+							</ClayButton>
+						</div>
+					</ClayCard>
+				</FocusTrap>
+			)}
+		</>
+	);
+};
+
+export const FocusOnSpecificElement = () => {
+	const [active, setActive] = useState(false);
+	const activateTrapButton = useRef<HTMLButtonElement>(null);
+	const lastButton = useRef<HTMLButtonElement>(null);
+
+	const onActivateFocusTrap = () => {
+		setActive(true);
+	};
+
+	const onDeactivateFocusTrap = () => {
+		setActive(false);
+
+		if (activateTrapButton.current) {
+			activateTrapButton.current.focus();
+		}
+	};
+
+	return (
+		<>
+			<ClayButton onClick={onActivateFocusTrap} ref={activateTrapButton}>
+				Activate trap
+			</ClayButton>
+			{active && (
+				<FocusTrap
+					active={active}
+					focusElementRef={lastButton}
+					onDeactivate={onDeactivateFocusTrap}
```

Comment:
I think I wouldn't need `active` and `onDeactivate` anymore since the conditional is done outside the component, to restore the focus, I would add a `useEffect` with the dep of `active`.

Reply:
In this case, there may be components like the Management Toolbar that need the props `active` inside the component. For example, on desktop, the management toolbar displays the following search form:

<img width="1218" alt="Screen Shot 2023-03-28 at 4 09 03 PM" src="https://user-images.githubusercontent.com/9701095/228271485-629a81a0-03d1-4312-b174-42d15cce8ee2.png">

This search form is the same as on mobile, which here is hidden by css classes. In this case the focus trap should wrap the search form, but we can't hide it because it would also hide on desktop (on desktop there shouldn't be focus trap either).

https://user-images.githubusercontent.com/9701095/228273513-b56c5989-3f21-4f4c-9e7b-b486bac16bfb.mov

<br></br>
About the `onDeactive` props is simply to be executed when the user presses the escape key. Do you think there may be another way to do this?

Reply:
Hmm I see, esse é um caso valido com certeza. 
>